.github/workflows/cmake.yml

```diff
@@ -38,6 +38,10 @@ jobs:
         if [ "${MATRIX_COMPILER}" = "clang" ]; then
           EXTRA_OPTS="-f"
         fi
+        if [ "${{ matrix.compiler }}" = "gcc" ]; then
+            ADD_FLAGS="-flto=$(nproc)"
```

Comment:
You can use auto here as that will tell gcc to communicate with a parent GNU Make and collaborate with whatever -j value was chosen there.

Unfortunately this doesn't work for Ninja, but it appears you're using the cmake default backend ("Unix Makefiles")?

Reply:
Yes, that's the way the MG devs have had it ever since I first stumbled on the project in 2016

Reply:
> Unfortunately this doesn't work for Ninja, but it appears you're using the cmake default backend ("Unix Makefiles")?

@eli-schwartz Any idea how to get it to work if someone specifies to use Ninja as a backend?

Reply:
Ninja doesn't provide a jobserver. There is nothing to do other than use single-threaded LTO or heuristically guess some number of cores yourself.

There is a PR proposing to implement a jobserver, we'll see how that goes. (Actually there are like 3 different competing PRs.)